### DISCUSSION TITLE
JS errors for elements in hidden (teleported) template

### DISCUSSION THREAD

**Discussion by simonferndriger** [alpinejs/alpine#3552] [1. Help]
When the button is pressed in [this pen](https://codepen.io/simonferndriger/pen/eYPGRwN?editors=1010), you'll get the JS error that `foo.bar` can't be found, because the element has been teleported. But shouldn't it still work? How can the error message be avoided?

**Comment by ekwoka** [comment]
So this is an error, that would be internal to x-teleport.

It is not destructive in any way, though.

It's basically just that the cascade of x-if removing the template of x-teleport to the x-teleport cleaning up the scope, and removing the `h1` is not happening all before the expressions in the `h1` are reevaluating as a result of the effect dependencies changing.

I'll look into it. but that's where the issue is.

**Reply by simonferndriger** [reply]
Any update on this? Since we are going into production now, having this additional JS errors makes it hard to tell them apart from real errors when tracking them with sentry... :)

**Reply by ekwoka** [reply]
It should go away, if you put an element between the two templates.
